### PR TITLE
Login & Registration Integration

### PR DESCRIPTION
**Pull request by Plootonix**
- Created screens for login and registration
- Created AuthContext and AuthProvider so app handles authentication changes
- Added styles for login and registration screens
- Logout button on main page with user greeting
- Exercise filter moved to bottom

### CODE REVIEW

File: GrizzFit/services/login.js [hunk 1]
```diff
@@ -2,5 +2,10 @@ import React from 'react';
 import { f, auth } from '../config/config';
 
 export const registerUser = () => {
```
Comment (itsDalcaN) [inline GrizzFit/services/login.js:4]: i don't believe we need this file anymore because of AuthProvider

File: GrizzFit/src/screens/Register.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+        // TODO: Create a new user document in Firestore and add their name to it
+    }
+
+    const navLogin = () => {
+        navigation.navigate("Login")
+    } 
+
+    return (
+        <KeyboardAvoidingView
```
Comment (itsDalcaN) [inline GrizzFit/src/screens/Register.js:29]: Not sure if we could get this KeyboardAvoidingView to work on Android too :/

File: GrizzFit/src/screens/LoginScreen.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    const {login} = useContext(AuthContext);
+
+    const navigation = useNavigation()
+
+    const navRegister = () => {
+        navigation.navigate("Register")
+    }
+
+    return (
+        <KeyboardAvoidingView
```
Comment (itsDalcaN) [inline GrizzFit/src/screens/LoginScreen.js:22]: Not sure if we could get this KeyboardAvoidingView to work on Android too :/

File: GrizzFit/src/screens/LoginScreen.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                    value={email}
+                    onChangeText={text => setEmail(text)}
+                    style={styles.input}
+                />
+                <TextInput
+                    placeholder="Password"
+                    value={password}
+                    onChangeText={text => setPassword(text)}
+                    style={styles.input}
+                    secureTextEntry
```
Comment (richardpluta) [inline GrizzFit/src/screens/LoginScreen.js:39]: So I was looking at these TextInputs and saw that there's a `textContentType` property you can set which changes the keyboard that appears and affects how the OS can autofill in these fields. 

You could do use the following for the different TextInputs you're using across these components:
```js
textContentType="name"
textContentType="emailAddress"
textContentType="password"
```

Comment (richardpluta) [reply GrizzFit/src/screens/LoginScreen.js:39]: Here's a list of all the possible content types for the TextInput:
```js
'none'
'URL'
'addressCity'
'addressCityAndState'
'addressState'
'countryName'
'creditCardNumber'
'emailAddress'
'familyName'
'fullStreetAddress'
'givenName'
'jobTitle'
'location'
'middleName'
'name'
'namePrefix'
'nameSuffix'
'nickname'
'organizationName'
'postalCode'
'streetAddressLine1'
'streetAddressLine2'
'sublocality'
'telephoneNumber'
'username'
'password'
'newPassword'
'oneTimeCode'
```

File: GrizzFit/src/screens/Register.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+                    onChangeText={text => setEmail(text)}
+                    style={styles.input}
+                />
+                <TextInput 
+                    placeholder="Password"
+                    value={password}
+                    onChangeText={text => setPassword(text)}
+                    style={styles.input}
+                    secureTextEntry
+                />
```
Comment (richardpluta) [inline GrizzFit/src/screens/Register.js:53]: Same comment about the TextInputs here.

**Review by richardpluta — DISMISSED** [review]
Gonna approve this but I think it would be worth addressing all the comments first.

File: GrizzFit/src/screens/Register.js (hunk 2 above)
Comment (richardpluta) [reply GrizzFit/src/screens/Register.js:29]: A quick glance at the docs seems to say that the `behavior` attribute should be set dynamically based on the OS detected. 

https://reactnative.dev/docs/keyboardavoidingview#example

Comment (richardpluta) [reply GrizzFit/src/screens/Register.js:29]: So this should look something like this:
```jsx
        <KeyboardAvoidingView
            style={styles.container}
            behavior={Platform.OS === "ios" ? "padding" : "height"}
        >
```
Just gotta import Platform.OS.

**Review by richardpluta — APPROVED** [review]

**Pull request merged**